precompiles/ArbOwner.go

```diff
@@ -157,3 +161,50 @@ func (con ArbOwner) ReleaseL1PricerSurplusFunds(c ctx, evm mech, maxWeiToRelease
 	}
 	return weiToTransfer, nil
 }
+
+func (con ArbOwner) SetChainConfig(c ctx, evm mech, serializedChainConfig []byte) error {
+	if c == nil {
+		return errors.New("nil context")
+	}
+	if c.txProcessor == nil {
+		return errors.New("uninitialized tx processor")
+	}
+	if c.txProcessor.MsgIsGasEstimation() {
+		var newConfig params.ChainConfig
+		err := json.Unmarshal(serializedChainConfig, &newConfig)
+		if err != nil {
+			return fmt.Errorf("invalid chain config, can't deserialize: %w", err)
+		}
+		if newConfig.ChainID == nil {
+			return errors.New("invalid chain config, missing chain id")
+		}
+		// TODO do we want to validate chain id? Is a change of chain id a valid operation?
+		chainId, err := c.State.ChainId()
+		if err != nil {
+			return fmt.Errorf("failed to get chain id from ArbOS state: %w", err)
+		}
+		if newConfig.ChainID.Cmp(chainId) != 0 {
+			return fmt.Errorf("invalid chain config, chain id mismatch, want: %v, have: %v", chainId, newConfig.ChainID)
+		}
+		oldSerializedConfig, err := c.State.ChainConfig()
+		if err != nil {
+			return fmt.Errorf("failed to get old chain config from ArbOS state: %w", err)
+		}
+		if bytes.Equal(oldSerializedConfig, serializedChainConfig) {
+			return errors.New("new chain config is the same as old")
+		}
+		var oldConfig params.ChainConfig
+		err = json.Unmarshal(oldSerializedConfig, &oldConfig)
+		if err != nil {
+			return fmt.Errorf("failed to deserialize old config: %w", err)
+		}
+		if err := oldConfig.CheckCompatible(&newConfig, newConfig.ArbitrumChainParams.GenesisBlockNum, 0); err != nil {
```

Comment:
- The old config might not be set (`len(oldSerializedConfig) == 0`), in which case we shouldn't deserialize or check the old config here because it'd fail
- We should CheckCompatible as of the block this call is being executed against: `evm.Context.BlockNumber` and `evm.Context.Time`